server/issue.go

```diff
@@ -571,6 +571,11 @@ func (s *Server) getPipelineFromIssue(ctx context.Context, issueCreate *api.Issu
 			return nil, err
 		}
 
+		// Validate the labels.
+		if err := s.setDatabaseLabels(ctx, m.Labels, &api.Database{Instance: instance} /* dummp database */, 0 /* dummp updaterID */, true /* validateOnly */); err != nil {
```

Comment:
If all we do here is to validate the label, it feels more natural to just call validateXXX here.

So instead of adding validateOnly flag to setDatabaseLabels, it's the other way around to keep setDatabaseLabels as is and extract the common method into validateXXX

wdty?

Reply:
I feel createPipelineFromIssue is also confusing because createXXX infers the method includes code to persist the resource (it's ok if it has a validateOnly flag to gate that logic). However, for this method, it's not "creating", it's just "building" the pipeline, it's more similar like the builder pattern, so I would feel calling it bulidXXX makes more sense.

Thus there is no need to pass the validate flag at all, since the buildXXX by its definition just builds the stuff. And if we look at this method, we do have api.ValidateAndGetDeploymentSchedule inside it conforming this semantics.

So I would still argue for creating a separate validate method here.

I know I am being insistency here because it's quite a core and complex piece and I try to make it more readable to other folks.

Reply:
Per offline discussion, will go with the current implementation and wait till the migration completes and revisit this.